_benchmark/user-guide/optimizing-benchmarks/performance-testing-best-practices.md

```diff
@@ -0,0 +1,277 @@
+    },
+    'system': {
+        'cpu_usage': '%',
+        'memory_used': 'bytes',
+        'disk_io': 'iops'
+    }
+}
+```
+
+### Metrics calculation
+
+OpenSearch Benchmark calculates metrics differently from traditional client-server systems. For detailed information on how metrics are calculated, see [Differences between OpenSearch Benchmark and a traditional client server system](https://opensearch.org/docs/latest/benchmark/user-guide/concepts/#differences-between-opensearch-benchmark-and-a-traditional-client-server-system).
+
+## Integration with OpenSearch Dashboards
+
+To integrate OpenSearch Benchmark results with OpenSearch Dashboards, you can perform one of the following:
+
+1. Configure OpenSearch Benchmark to store results in OpenSearch.
+2. Create index patterns in OpenSearch Dashboards for the benchmark results.
+3. Create visualizations and dashboards to analyze the benchmark data.
+
+
+## Common pitfalls
+
+When conducting performance tests with OpenSearch Benchmark, it's important to be aware of common pitfalls that can lead to inaccurate or misleading results.
+
+### Warmup intervals
+
+Proper warmup is crucial for accurate performance testing. Without an adequate warmup period, your test results may be skewed by initial system instabilities or caching effects.
+
+Don't run tests without a warmup period
+
+Instead, always include an adequate warmup period in your tests. This allows the system to reach a steady state before measurements begin. The following example gives a `geonames` run a warmup period of `300s`.
+
+```python
+# DO: Include adequate warmup
+opensearch-benchmark run --workload=geonames --warmup-time-period=300
+```
+
+The appropriate warmup time can vary depending on your specific workload and system configuration. Start with at least 5 minutes (300 seconds) and adjust as needed based on your observations.
+
+### Compare results from different environments
+
+One of the most common mistakes in performance testing is comparing results from different environments. Results obtained from a laptop or development machine are not comparable to those from a production server due to differences in hardware, network conditions, and other environmental factors.
+
+Instead, make that all comparisons are made within the same or identical environments. If you need to compare different configurations, make sure to change only one variable at a time while keeping the environment consistent.
+
+### Document environment details
+
+Proper documentation of your test environment is crucial for reproducibility and accurate analysis. Without detailed environment information, it becomes difficult to interpret results or reproduce tests in the future.
+
+Don't omit environment details from your test reports:
+
+Instead, always document comprehensive details about your test environment. This should include hardware specifications, software versions, and any relevant configuration settings. as shown in the following example:
+
```

Comment:
Is this an example of using opensearch-benchmark in a script? If so, we can improve this and also mention that this example is from a user who is using opensearch-benchmark in a script to easily reproduce runs.